server/src/test/java/org/opensearch/common/network/NetworkModuleTests.java

```diff
@@ -160,6 +193,60 @@ public Map<String, Supplier<HttpServerTransport>> getHttpTransports(
         expectThrows(IllegalStateException.class, () -> newModule.getHttpServerTransportSupplier());
     }
 
+    public void testRegisterSecureTransport() {
+        Settings settings = Settings.builder().put(NetworkModule.TRANSPORT_TYPE_KEY, "custom-secure").build();
+        Supplier<Transport> custom = () -> null; // content doesn't matter we check reference equality
+        NetworkPlugin plugin = new NetworkPlugin() {
+            @Override
+            public Map<String, Supplier<Transport>> getSecureTransports(
+                Settings settings,
+                ThreadPool threadPool,
+                PageCacheRecycler pageCacheRecycler,
+                CircuitBreakerService circuitBreakerService,
+                NamedWriteableRegistry namedWriteableRegistry,
+                NetworkService networkService,
+                SecureTransportSettingsProvider secureTransportSettingsProvider,
+                Tracer tracer
+            ) {
+                return Collections.singletonMap("custom-secure", custom);
+            }
+        };
+        NetworkModule module = newNetworkModule(settings, null, List.of(secureTransportSettingsProvider), plugin);
+        assertSame(custom, module.getTransportSupplier());
+    }
+
+    public void testRegisterSecureHttpTransport() {
+        Settings settings = Settings.builder()
+            .put(NetworkModule.HTTP_TYPE_SETTING.getKey(), "custom-secure")
+            .put(NetworkModule.TRANSPORT_TYPE_KEY, "local")
+            .build();
+        Supplier<HttpServerTransport> custom = FakeHttpTransport::new;
+
+        NetworkModule module = newNetworkModule(settings, null, List.of(secureTransportSettingsProvider), new NetworkPlugin() {
+            @Override
+            public Map<String, Supplier<HttpServerTransport>> getSecureHttpTransports(
+                Settings settings,
+                ThreadPool threadPool,
+                BigArrays bigArrays,
+                PageCacheRecycler pageCacheRecycler,
+                CircuitBreakerService circuitBreakerService,
+                NamedXContentRegistry xContentRegistry,
+                NetworkService networkService,
+                HttpServerTransport.Dispatcher requestDispatcher,
+                ClusterSettings clusterSettings,
+                SecureTransportSettingsProvider secureTransportSettingsProvider,
+                Tracer tracer
+            ) {
+                return Collections.singletonMap("custom-secure", custom);
+            }
+        });
+        assertSame(custom, module.getHttpServerTransportSupplier());
+
+        settings = Settings.builder().put(NetworkModule.TRANSPORT_TYPE_KEY, "local").build();
+        NetworkModule newModule = newNetworkModule(settings, null);
+        expectThrows(IllegalStateException.class, () -> newModule.getHttpServerTransportSupplier());
```

Comment:
Can we inspect the expection in more detail - I am having trouble understanding why this exception is expected

Reply:
It checks for transport key that does not exist, I could actually remove this part cuz it is also tested by `testRegisterHttpTransport()`